src/main/java/com/microsoft/azure/functions/worker/broker/JavaMethodExecutor.java

```diff
@@ -42,24 +50,69 @@ public JavaMethodExecutor(FunctionMethodDescriptor descriptor, Map<String, Bindi
         for (Map.Entry<String, BindingInfo> entry : bindingInfos.entrySet()) {
             this.bindingDefinitions.put(entry.getKey(), new BindingDefinition(entry.getKey(), entry.getValue()));
         }
+        resolveFunctionFactory();
     }
-    
+
+    // TODO remove once I figure out how to see logging in local docker
+    private static Logger fileLogger;
+    static {
+        fileLogger = Logger.getAnonymousLogger();
+        fileLogger.setUseParentHandlers(false);
+        FileHandler fh = null;
+        try {
+            fh = new FileHandler("java_worker.log");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        fh.setFormatter(new SimpleFormatter());
+        fh.setLevel(Level.ALL);
+        fileLogger.addHandler(fh);
+        fileLogger.info("INITIALIZED LOGGER");
+
+    }
+    private Logger getLogger() {
+        return fileLogger;
+    }
+
     Map<String, BindingDefinition> getBindingDefinitions() { return this.bindingDefinitions; }
     
     public ParameterResolver getOverloadResolver() { return this.overloadResolver; }
 
     void execute(BindingDataStore dataStore) throws Exception {
         Object retValue = this.overloadResolver.resolve(dataStore)
             .orElseThrow(() -> new NoSuchMethodException("Cannot locate the method signature with the given input"))
-            .invoke(() -> this.containingClass.newInstance());
+            .invoke(() -> createFunctionInstance());
         dataStore.setDataTargetValue(BindingDataStore.RETURN_NAME, retValue);
     }
-    
+
+    private Object createFunctionInstance() throws Exception {
+        if (functionFactory == null) {
+            return this.containingClass.newInstance();
+        }
+        return functionFactory.invoke(null, containingClass);
+    }
+
     Class<?> getContainingClass(String className, ClassLoaderProvider classLoaderProvider) throws ClassNotFoundException {
         ClassLoader classLoader = classLoaderProvider.getClassLoader();
         return Class.forName(className, true, classLoader);
     }
 
+    private void resolveFunctionFactory() throws IOException, ClassNotFoundException, NoSuchMethodException {
+        ClassLoader cl = containingClass.getClassLoader();
+        InputStream is = cl.getResourceAsStream("META-INF/service/com.microsoft.azure.functions.FunctionFactory");
+        if (is == null) {
+            getLogger().info("resolveFunctionFactory could not find descriptor");
```

Comment:
need to be changed to [systemLogger](https://github.com/Azure/azure-functions-java-worker/blob/8845d98d1741586a15ed72db3e1442a7f6eaad38/src/main/java/com/microsoft/azure/functions/worker/WorkerLogManager.java#L11) to merge.